client/src/app/components/tool-container/tool-header/tool-header.component.ts

```diff
@@ -0,0 +1,63 @@
+import {
+    Component,
+    OnInit,
+    EventEmitter,
+    Input,
+    Output
+} from "@angular/core";
+import {NgSelectOption} from "@angular/common";
+import {ViewMode} from "../tool-container.component";
+
+require("./tool-header.component.scss");
+
+@Component({
+    selector: "tool-header",
+    directives: [NgSelectOption],
+    template: `
+        <header>
+            <select id="revisionSelect" name="rev" [ngModel]="selectedRevision" (change)="onRevisionChange($event)">
+                <option *ngFor="let revision of revisions" [value]="revision">{{revision}}</option>
+            </select>
+                            <span id="guiJsonButtons">
```

Comment:
Avoid using ids for components that are reusable. According to [the MDN](https://developer.mozilla.org/en/docs/Web/HTML/Global_attributes/id): _the id global attribute defines a unique identifier (ID) which must be unique in the whole document_. It's impossible to ensure that an id in a component will remain unique across the whole document.